sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -177,9 +177,17 @@ object SQLExecution {
     val sc = sparkSession.sparkContext
     val localProps = Utils.cloneProperties(sc.getLocalProperties)
     Future {
+      val originalContext = SparkSession.getActiveSession
+      val originalLocalProps = Utils.cloneProperties(sc.getLocalProperties)
```

Comment:
do we need to clone?

Reply:
We can assure the `LocalProps` is used in current thread here, `getLocalProperties` is enough, done in a33154c, thanks.

Reply:
I think clone is required. Please read 
https://github.com/apache/spark/pull/27267#discussion_r369023941

Reply:
> I think clone is required.

The two scenarios are different, the first one needs clone since we want a `snapshot` of local properties during submit. But for this one, we need to reset by the variable self. If use clone here, changes in other places might be dropped.

> Also i dont see why we need to reset.

For the function added in SQLExecution here, I think we need to clean up all the side effects as far as possible, even though it works in this case.